source/common/tcp_proxy/tcp_proxy.h

```diff
@@ -128,16 +127,22 @@ class Config {
     std::string cluster_name_;
   };
 
-  struct WeightedClusterEntry {
+  class WeightedClusterEntry {
+  public:
     WeightedClusterEntry(const envoy::config::filter::network::tcp_proxy::v2::TcpProxy::
                              WeightedCluster::ClusterWeight& config);
 
+    const std::string& clusterName() const { return cluster_name_; }
+    uint64_t clusterWeight() const { return cluster_weight_; }
+
+  private:
     const std::string cluster_name_;
     const uint64_t cluster_weight_;
   };
+  typedef std::shared_ptr<WeightedClusterEntry> WeightedClusterEntrySharedPtr;
```

Comment:
I think this can be a unique_ptr instead of shared_ptr. I don't think these are ever shared.

Reply:
Fixed in af356c1.